lucille-plugins/lucille-pinecone/src/test/java/com/kmwllc/lucille/pinecone/util/PineconeManagerTest.java

```diff
@@ -0,0 +1,56 @@
+package com.kmwllc.lucille.pinecone.util;
+
+import static org.mockito.Mockito.*;
+import static org.junit.Assert.*;
+
+import io.pinecone.clients.Pinecone;
+import org.junit.Test;
+import org.mockito.MockedConstruction;
+import org.mockito.Mockito;
+
+public class PineconeManagerTest {
+
+  @Test
+  public void testReuseClient() throws Exception {
+    Pinecone mockClient = mock(Pinecone.class);
+    Pinecone differentClient = mock(Pinecone.class);
+    try(MockedConstruction<Pinecone.Builder> builder = Mockito.mockConstruction(Pinecone.Builder.class,(mock,context)-> {
+      // will return mockClient first time is called, then return a different client when called a second time
+      when(mock.build()).thenReturn(mockClient).thenReturn(differentClient);
+    })) {
+      // test that "sameClient" is retrieving the same client object, and that .build was only called trice
```

Comment:
Should be "thrice" in the above comment but maybe we should change it to "three times". Only suggesting because I misread it as twice at first and was surprised when the test asserted it was called 3 times.